homeassistant/components/rfxtrx/__init__.py

```diff
@@ -236,25 +241,66 @@ def send(call):
     hass.services.async_register(DOMAIN, SERVICE_SEND, send, schema=SERVICE_SEND_SCHEMA)
 
 
+async def async_setup_platform_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    supported: Callable[[rfxtrxmod.RFXtrxEvent], bool],
+    constructor: Callable[
+        [rfxtrxmod.RFXtrxEvent, rfxtrxmod.RFXtrxEvent | None, DeviceTuple, dict],
+        list[Entity],
+    ],
+):
+    """Set up config entry."""
+    entry_data = config_entry.data
+    device_ids: set[DeviceTuple] = set()
+
+    # Add entities from config
+    entities = []
+    for packet_id, entity_info in entry_data[CONF_DEVICES].items():
+        if (event := get_rfx_object(packet_id)) is None:
+            _LOGGER.error("Invalid device: %s", packet_id)
+            continue
+        if not supported(event):
+            continue
+
+        device_id = get_device_id(
+            event.device, data_bits=entity_info.get(CONF_DATA_BITS)
+        )
+        if device_id in device_ids:
+            continue
+        device_ids.add(device_id)
+
+        entities.extend(constructor(event, None, device_id, entity_info))
+
+    async_add_entities(entities)
+
+    # If automatic add is on, hookup listener
+    if entry_data[CONF_AUTOMATIC_ADD]:
```

Comment:
You can save some indent by reversing the condition, return and outdent below

Reply:
I have a follow-up pull that drops the check anyway. I'll handle it then.